File: corpus/cs1/0007-botas-perdidas/solutions/a-optimal/solve_a.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int n;
    while (cin >> n) {
        vector<int> left(65, 0);
        vector<int> right(65, 0);

        for (int i = 0; i < n; ++i) {
            int size;
            char side;
            cin >> size >> side;
            if (side == 'E') {
                left[size]++;
            } else {
                right[size]++;
            }
        }

        int pairs = 0;
        for (int i = 30; i <= 60; ++i) {
            pairs += min(left[i], right[i]);
        }
        cout << pairs << endl;
    }
    return 0;
}

File: corpus/cs1/0007-botas-perdidas/solutions/a-optimal/SolveA.java
import java.util.Scanner;

public class SolveA {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        while (scanner.hasNextInt()) {
            int n = scanner.nextInt();
            int[] left = new int[65];
            int[] right = new int[65];

            for (int i = 0; i < n; i++) {
                int size = scanner.nextInt();
                String side = scanner.next().trim().toUpperCase();
                if (side.equals("E")) {
                    left[size]++;
                } else {
                    right[size]++;
                }
            }

            int pairs = 0;
            for (int i = 30; i <= 60; i++) {
                pairs += Math.min(left[i], right[i]);
            }
            System.out.println(pairs);
        }
        scanner.close();
    }
}
